regression/esbmc/01_pthread63/main.c

```diff
@@ -0,0 +1,30 @@
+#include <pthread.h>
+#define N 2
+
+int a[N], i, j;
```

Comment:
Why to use this define here?

Reply:
This is intentional :-)

I wanted to create an example to produce lots of interleavings so that I could double-check the number of interleavings that could be pruned.

Since `a[N]`, `i`, `j` are global variables and accessed by different threads, lots of interleavings are produced here.

ESBMC tries to evaluate the array accesses to check whether they overlap; as a result, we can prune the number of interleavings if the threads access different positions of the array.

Reply:
This is intentional :-)

I wanted to create an example to produce lots of interleavings so that I could double-check the number of interleavings that could be pruned.

Since `a[N]`, `i`, `j` are global variables and accessed by different threads, lots of interleavings are produced here.

ESBMC tries to evaluate the array accesses to check whether they overlap; as a result, we can prune the number of interleavings if the threads access different positions of the array.